File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        final_result = None
        count = 0

        for i in nums:
            if count == 0:
                final_result = i
            if final_result == i:
                count += 1
            else:
                count -= 1
        return final_result

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int final_input = 0;
        int count = 0;

        for(int i=0;i<nums.length;i++){
            if(count == 0){
                final_input = nums[i];
            }
            if (final_input == nums[i]){
                count += 1;
            }
            else{
                count -= 1;
            }
        }
        return final_input;
    }
}